tests/rptest/tests/offset_for_leader_epoch_test.py

```diff
@@ -188,14 +188,30 @@ def have_all_offsets():
 
         rpk = RpkTool(self.redpanda)
 
+        def get_offsets_and_epochs():
+            offsets = []
+
+            def refresh():
+                result = rpk.describe_topic(topic.name)
+                offsets.clear()
+                offsets.extend(result)
+
+            def all_offsets_valid():
+                refresh()
+                return all([p.high_watermark >= 0 for p in offsets])
+
+            wait_until(all_offsets_valid, 30, 1)
+
+            return offsets
```

Comment:
fyi: this is probably a use case for `wait_until_result` for combining wait_unitl with returning a value from the predicate.